hooks/useUpdateWeb3Store/index.ts

```diff
@@ -20,6 +20,11 @@ export const useUpdateWeb3Store: () => void = () => {
         }
     }, []);
 
+    // set default provider if we need to
+    useEffect(() => {
+        setDefaultProvider();
+    }, []);
+
```

Comment:
not exactly sure but I assume these two `useEffects` will get fired synchronously and in order they appear here. does it matter that we are setting the wallet before the default provider?

Reply:
yeah we actually want this, since then the preferred provider gets set if it exists (preferred provider being the injected one through onboard)